src/Core/Resolvers/MsSqlQueryExecutor.cs

```diff
@@ -217,9 +217,9 @@ public override string GetSessionParamsQuery(HttpContext? httpContext, IDictiona
             foreach ((string claimType, string claimValue) in sessionParams)
             {
                 string paramName = $"{SESSION_PARAM_NAME}{counter.Next()}";
-                parameters.Add(paramName, new(claimValue));
+                parameters.Add(paramName, new(claimValue));          
                 // Append statement to set read only param value - can be set only once for a connection.
-                string statementToSetReadOnlyParam = "EXEC sp_set_session_context " + $"'{claimType}', " + paramName + ", @read_only = 1;";
+                string statementToSetReadOnlyParam = "EXEC sp_set_session_context " + $"'{claimType}', " + paramName + ", @read_only = 0;";
```

Comment:
curious, did you experience  - your mutation operations being blocked because of this setting?

Reply:
Yes, the SQL Endpoint did reject the second query as described in https://github.com/Azure/data-api-builder/issues/2341